### PR TITLE
Add vpc_flow_log table. closes #103

### LINKED ISSUE #103: Add support to query, whether flow log associated with VPC in alicloud_vpc.
it will be great to query to know if VPC flow log is associated or not for each VPCs

Application table
https://hub.steampipe.io/plugins/turbot/alicloud/tables/alicloud_vpc

Ref
CIS 1.0.0  - 3.3 Ensure VPC flow logging is enabled in all VPCs (Manual)
                   2.5 Ensure virtual network flow log service is enabled (Manual)

image

### PR DESCRIPTION
**Pull request by bigdatasourav**
There is issue with flow log creation, discussed with Raj , on hold for now.

### CODE REVIEW

**Comment by bigdatasourav** [comment]
as discussed -
```
Error: [ERROR] terraform-provider-alicloud/alicloud/resource_alicloud_vpc_flow_log.go:126: Resource alicloud_vpc_flow_log CreateFlowLog Failed!!! [SDK alibaba-cloud-sdk-go ERROR]:
SDKError:
   Code: OperationUnsupported.actionError
   Message: code: 400, This action is not support. requestid: FAF4CBA6-357C-4A19-94F8-B78F4B9D6B97
   Data: {"Code":"OperationUnsupported.action","HostId":"vpc.aliyuncs.com","Message":"This action is not support.","Recommend":"https://error-center.aliyun.com/status/search?Keyword=OperationUnsupported.action\u0026source=PopGw","RequestId":"FAF4CBA6-357C-4A19-94F8-B78F4B9D6B97"}

File: alicloud/table_alicloud_vpc_flow_log.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+		return nil, err
+	}
+
+	var name string
+	if h.Item != nil {
+		log := h.Item.(vpc.FlowLog)
+		name = log.FlowLogName
+	} else {
+		name = d.KeyColumnQuals["name"].GetStringValue()
+	}
```
Comment (Subhajit97) [inline alicloud/table_alicloud_vpc_flow_log.go:160]: Do we need this check?

File: alicloud/table_alicloud_vpc_flow_log.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+	return nil, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func vpcFlowLogAkas(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	i := h.Item.(vpc.FlowLog)
+	// Get project details
```
Comment (Subhajit97) [inline alicloud/table_alicloud_vpc_flow_log.go:182]: Add new line before comment
```suggestion
	
	// Get project details
```

File: alicloud/table_alicloud_vpc_flow_log.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+	}
+
+	if response.FlowLogs.FlowLog != nil && len(response.FlowLogs.FlowLog) > 0 {
+		return response.FlowLogs.FlowLog[0], nil
+	}
+
+	return nil, nil
+}
+
+//// TRANSFORM FUNCTIONS
```
Comment (Subhajit97) [inline alicloud/table_alicloud_vpc_flow_log.go:178]:
```suggestion
```

File: alicloud/table_alicloud_vpc_flow_log.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+	// Get project details
+	commonData, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+	commonColumnData := commonData.(*alicloudCommonColumnData)
+	accountID := commonColumnData.AccountID
+	return []string{"acs:vpc:" + i.RegionId + ":" + accountID + ":flow-log/" + i.FlowLogId}, nil
+}
+
```
Comment (Subhajit97) [inline alicloud/table_alicloud_vpc_flow_log.go:191]:
```suggestion

//// TRANSFORM FUNCTIONS

```

File: docs/tables/alicloud_vpc_flow_log.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  flow_log_id,
+  status,
+  resource_type,
+  resource_id
+from
+  alicloud_vpc_flow_log;
+```
+
+
+### List flow logs where resource type is VPC.
```
Comment (Subhajit97) [inline docs/tables/alicloud_vpc_flow_log.md:21]:
```suggestion
### List flow logs where resource type is VPC
```

File: alicloud-test/tests/alicloud_vpc_flow_log/variables.tf [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  flow_log_name  = var.resource_name
+  status         = "Active"
+}
+
+
+
+output "resource_id" {
+  value = alicloud_vpc_flow_log.named_test_resource.id
+}
+
```
Comment (Subhajit97) [inline alicloud-test/tests/alicloud_vpc_flow_log/variables.tf:55]:
```suggestion
```

File: alicloud-test/tests/alicloud_vpc_flow_log/variables.tf [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  default = "vpc-flow-log-for-vpc"
+}
+
+variable "project_name" {
+  default = "vpc-flow-log-for-vpc"
+}
+
+resource "alicloud_vpc" "named_test_resource" {
+  cidr_block = "192.168.0.0/24"
+  name       = var.resource_name
```
Comment (Subhajit97) [inline alicloud-test/tests/alicloud_vpc_flow_log/variables.tf:36]: From alicloud provider `1.119.0`, `name` attribute is deprecated.
Use `vpc_name` instead.
```suggestion
  vpc_name       = var.resource_name
```

File: alicloud-test/tests/alicloud_vpc_flow_log/variables.tf [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  }
+}
+
+variable "log_store_name" {
+  default = "vpc-flow-log-for-vpc"
+}
+
+variable "project_name" {
+  default = "vpc-flow-log-for-vpc"
+}
```
Comment (Subhajit97) [inline alicloud-test/tests/alicloud_vpc_flow_log/variables.tf:32]: Can't we pass these values directly in `alicloud_vpc_flow_log` plan?

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@bigdatasourav Can you please remove `package-lock.json` file change from this commit?
Make suggested changes, and also add the add the outputs of the queries added in docs.

File: alicloud/table_alicloud_vpc_flow_log.go [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+	request := vpc.CreateDescribeFlowLogsRequest()
+	request.Scheme = "https"
+
+	response, err := client.DescribeFlowLogs(request)
+	if err != nil {
+		plugin.Logger(ctx).Error("alicloud_vpc_flow_log.listVpcFlowLogs", "query_error", err, "request", request)
+		return nil, err
+	}
+	for _, i := range response.FlowLogs.FlowLog {
+		plugin.Logger(ctx).Warn("alicloud_vpc.listVpc", "item", i)
```
Comment (Subhajit97) [inline alicloud/table_alicloud_vpc_flow_log.go:136]:
```suggestion
```

File: alicloud/plugin.go [hunk 10]
```diff
@@ -45,6 +45,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"alicloud_vpc_vpn_customer_gateway": tableAlicloudVpcVpnCustomerGateway(ctx),
 			"alicloud_vpc_vpn_gateway":          tableAlicloudVpcVpnGateway(ctx),
 			"alicloud_vpc_vswitch":              tableAlicloudVpcVSwitch(ctx),
+			"alicloud_vpc":                      tableAlicloudVpc(ctx),
 			"alicloud_ecs_key_pair":             tableAlicloudEcskeyPair(ctx),
```
Comment (Subhajit97) [inline alicloud/plugin.go:49]: Sort the list

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@bigdatasourav There are few changes required. Please see comments.
Also can you please add the outputs of the example queries?

**Comment by cbruno10** [comment]
Closing PR as this feature is currently blocked, will re-open once it's ready.

**Pull request closed without merging**